sfge/src/main/java/com/salesforce/rules/ops/boundary/SharingPolicyBoundary.java

```diff
@@ -0,0 +1,130 @@
+     */
+    @Override
+    public UserClassVertex getBoundaryItem() {
+        return this.boundaryItem;
+    }
+
+    /**
+     * Try to find the applicable ancestor to the boundary item. The applicable ancestor is defined
+     * as follows:
+     *
+     * <ul>
+     *   <li>If the boundary item has an explicitly defined sharing policy, the applicable ancestor
+     *       will be the boundary item.
+     *   <li>If the boundary item has no explicitly defined sharing policy, the applicable ancestor
+     *       will be the nearest ancestor with an explicitly defined sharing policy.
+     *   <li>If the boundary item has no explicitly defined sharing policy and no ancestors that
+     *       have an explicitly defined sharing policy, the applicable ancestor will be the
+     *       supermost {@link UserClassVertex} ancestor.
+     *   <li>If the boundary item has no explicitly defined sharing policy and no ancestors, the
+     *       applicable ancestor will be the boundary item.
+     *
+     * @return a {@link UserClassVertex} that is the applicable ancestor of the boundary item.
+     */
+    public UserClassVertex getApplicableAncestorPolicyVertex() {
+        // save the work of querying the graph until it's required
+        if (this.applicableAncestorVertex == null) {
+            this.applicableAncestorVertex =
+                    findAncestorWithExplicitSharingPolicy(this.boundaryItem).orElse(boundaryItem);
+        }
+        return this.applicableAncestorVertex;
+    }
+
+    /**
+     * Attempt to find an applicable ancestor of <code>child</code> that has an explicitly defined
+     * sharing policy (with, without, or inherited sharing). See {@link
+     * #getApplicableAncestorPolicyVertex()}
+     *
+     * @param child a {@link UserClassVertex} for which we are attempting to find an ancestor with
+     *     an explicitly defined sharing policy
+     * @return An optional of a {@link UserClassVertex} if an ancestor with an explicitly defined
+     *     sharing policy is found. Empty if there are no such ancestors (or no ancestors at all)
+     */
+    private Optional<UserClassVertex> findAncestorWithExplicitSharingPolicy(UserClassVertex child) {
+        if (!child.getSharingPolicy().equals(SharingPolicy.OMITTED_DECLARATION)) {
+            // if there is an explicitly defined sharing policy, return the child.
+            // inherited sharing falls under this category because it means we must look at the
+            // calling classes to find the sharing policy, not the child's ancestors.
+            return Optional.of(child);
+        } else {
+            // query the graph to recursively find all the ancestors of child
+            List<UserClassVertex> vertices =
+                    SFVertexFactory.loadVertices(
+                            g,
+                            g.V(child.getId())
+                                    .union(
+                                            __.identity(),
```

Comment:
Using `__.identity()` will cause `child` itself to be included in this array, which seems redundant. Does removing this clause break your tests?